unified-runtime/source/adapters/level_zero/v2/context.cpp

```diff
@@ -126,8 +139,39 @@ void ur_context_handle_t_::removeUsmPool(ur_usm_pool_handle_t hPool) {
   usmPoolHandles.remove(hPool);
 }
 
-const std::vector<ur_device_handle_t> &
-ur_context_handle_t_::getP2PDevices(ur_device_handle_t hDevice) const {
+void ur_context_handle_t_::addResidentDevice(ur_device_handle_t hDevice, ur_device_handle_t newPeerDevice) {
+  std::scoped_lock<ur_shared_mutex> lock(Mutex);
+  auto & pDevices = p2pAccessDevices.at(hDevice->Id);
+
+  assert(0 = std::count_if(
+    std::begin(pDevices), std::end(pDevices),
+        [&](const auto pDevice) { return newPeerDevice->Id == pDevice->Id; }));
```

Comment:
[cuCtxEnablePeerAccess](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__PEER__ACCESS.html#group__CUDA__PEER__ACCESS_1g0889ec6728e61c05ed359551d67b3f5a) returns `CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED`. Could we have this error mapped to a UR error? It would help make the peer-to-peer SYCL API more consistent (See https://github.com/intel/llvm/pull/19787.)